### PR TITLE
misc fixes

### PR DESCRIPTION
**Pull request by Nivaturimika**

### CODE REVIEW

File: src/ai/ai.cpp [hunk 1]
```diff
@@ -2758,56 +2758,69 @@ bool will_accept_peace_offer_value(sys::state& state,
 	int32_t war_duration, bool contains_sq) {
 	bool is_attacking = !offer_from_attacker;
 
-	auto overall_score = primary_warscore;
+	auto overall_score = military::primary_warscore(state, w);
```
Comment (schombert) [inline src/ai/ai.cpp:2761]: why are you recalculating this rather than using the value passed in?

File: src/ai/ai.cpp [hunk 2]
```diff
@@ -2758,56 +2758,69 @@ bool will_accept_peace_offer_value(sys::state& state,
...
 
-	auto personal_score_saved = personal_po_value - potential_peace_score_against;
+	auto personal_score_saved = target_personal_po_value - potential_peace_score_against;
 
 	if((prime_attacker == n || prime_defender == n) && (prime_attacker == from || prime_defender == from)) {
 		if(overall_score <= -50 && overall_score <= overall_po_value * 2)
 			return true;
 
-		if(concession && my_side_peace_cost <= overall_po_value)
+		auto war_duration = state.current_date.value - state.world.war_get_start_date(w).value;
```
Comment (schombert) [inline src/ai/ai.cpp:2781]: again, why is this being recalculated?

File: src/ai/ai.cpp [hunk 3]
```diff
@@ -2758,56 +2758,69 @@ bool will_accept_peace_offer_value(sys::state& state,
...
 			return false;
 		}
 		float willingness_factor = float(war_duration - 365) * 10.f / 365.0f;
 		if(overall_score >= 0) {
-			if(concession && (overall_score * 2 - overall_po_value - willingness_factor) < 0)
+			if(concession && ((overall_score * 2 - overall_po_value - willingness_factor) < 0))
 				return true;
 		} else {
-			if(overall_score <= overall_po_value && (overall_score / 2 - overall_po_value - willingness_factor) < 0)
+			if((overall_score - willingness_factor) <= overall_po_value && (overall_score / 2 - overall_po_value - willingness_factor) < 0)
```
Comment (schombert) [inline src/ai/ai.cpp:2792]: do these changes match the function below?

File: src/ai/ai.cpp [hunk 4]
```diff
@@ -2758,56 +2758,69 @@ bool will_accept_peace_offer_value(sys::state& state,
...
 				return true;
 		}
 
 	} else if((prime_attacker == n || prime_defender == n) && concession) {
+		auto scoreagainst_me = military::directed_warscore(state, w, from, n);
```
Comment (schombert) [inline src/ai/ai.cpp:2797]: again, why is this being recalculated?

File: src/ai/ai.cpp [hunk 5]
```diff
@@ -2758,56 +2758,69 @@ bool will_accept_peace_offer_value(sys::state& state,
...
 
 		if(scoreagainst_me > 50)
 			return true;
 
+		int32_t my_side_against_target = 0;
+		for(auto wg : state.world.war_get_wargoals_attached(w)) {
+			if(wg.get_wargoal().get_target_nation() == from) {
+				auto wg_value = military::peace_cost(state, w, wg.get_wargoal().get_type(), wg.get_wargoal().get_added_by(), n, wg.get_wargoal().get_secondary_nation(), wg.get_wargoal().get_associated_state(), wg.get_wargoal().get_associated_tag());
+
+				my_side_against_target += wg_value;
```
Comment (schombert) [inline src/ai/ai.cpp:2807]: see above

File: src/ai/ai.cpp (hunk 3 above)
Comment (Nivaturimika) [reply src/ai/ai.cpp:2787]: yes

File: src/ai/ai.cpp [hunk 6]
```diff
@@ -2758,52 +2758,57 @@ bool will_accept_peace_offer_value(sys::state& state,
...
 
-	auto personal_score_saved = personal_po_value - potential_peace_score_against;
+	auto personal_score_saved = target_personal_po_value - potential_peace_score_against;
 
 	if((prime_attacker == n || prime_defender == n) && (prime_attacker == from || prime_defender == from)) {
 		if(overall_score <= -50 && overall_score <= overall_po_value * 2)
 			return true;
 
-		if(concession && my_side_peace_cost <= overall_po_value)
+		if(concession && (is_attacking ? military::attacker_peace_cost(state, w) : military::defender_peace_cost(state, w)) <= overall_po_value)
```
Comment (schombert) [inline src/ai/ai.cpp:2777]: this also appears to be recalculating a value passed to the function

**Pull request merged**